### PR TITLE
doc: add app migration doc

### PR DESCRIPTION
**Pull request by xiang90**
@heyitsanthony @gyuho @hongchaodeng @joshix @robszumski @jakedt @timothysc

Fix https://github.com/coreos/etcd/issues/4969

### CODE REVIEW

File: Documentation/op-guide/v2-migration.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Migrate applications from using API v2 to API v3
+
+The data store v2 is still accessible from the API v2 after upgrading to etcd3. Thus, it will work as before and require no application changes. With etcd 3, applications use the new grpc API v3 to access the mvcc store, which provides more features and improved performance. The mvcc store and the old store v2 are separate and isolated; writes to the store v2 will not affect the mvcc store and, similarly, writes to the mvcc store will not affect the store v2.
+
+Migrating an application from the API v2 to the API v3 involves two steps: 1) migrate the client library and, 2) migrate the data. If the application can rebuild the data, then migrating the data is unnecessary.
+
+## Migrate client library
+
+API v3 is different from API v3, thus application developers need to use a new client library to send requests to etcd API v3. The documentation of the client v3 is available at https://godoc.org/github.com/coreos/etcd/clientv3.
```
Comment (gyuho) [inline Documentation/op-guide/v2-migration.md]: `API v2 is different from API v3`?

File: Documentation/op-guide/v2-migration.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## Migrate client library
+
+API v3 is different from API v3, thus application developers need to use a new client library to send requests to etcd API v3. The documentation of the client v3 is available at https://godoc.org/github.com/coreos/etcd/clientv3. 
+
+There are some notable differences between API v2 and API v3:
+
+- Transaction: In v3, etcd provides capability to do multi-key transaction with conditional flow. Applications need to use transactions instead of Compare-And-Swap.
+
+- Flat key space: There is no directory in API v3. For example, "/a/b/c" is a key. If you want to get all keys under a prefix, you can do a range query.
```
Comment (heyitsanthony) [inline Documentation/op-guide/v2-migration.md]: - Flat key space: There are no directories in API v3, only keys. For example, "/a/b/c/" is a key. Range queries support getting all keys matching a given prefix.

File: Documentation/op-guide/v2-migration.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+- Compacted response. Operations like `Delete` doesn’t return previous value anymore. If you want to get the value being deleted, you can do a transaction that gets the previous value and deletes the key.
```
Comment (heyitsanthony) [inline Documentation/op-guide/v2-migration.md]: Compacted responses. Operations like `Delete` no longer return previous values. To get the deleted value, a transaction can be used to atomically delete the key and get its value.

**Comment by xiang90** [comment]
@heyitsanthony All fixed.

File: Documentation/op-guide/v2-migration.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+Migrating an application from the API v2 to the API v3 involves two steps: 1) migrate the client library and, 2) migrate the data. If the application can rebuild the data, then migrating the data is unnecessary.
+
+## Migrate client library
+
+API v3 is different from API v2, thus application developers need to use a new client library to send requests to etcd API v3. The documentation of the client v3 is available at https://godoc.org/github.com/coreos/etcd/clientv3. 
+
+There are some notable differences between API v2 and API v3:
+
+- Transaction: In v3, etcd provides capability to do multi-key transaction with conditional flow. Applications need to use transactions instead of Compare-And-Swap.
```
Comment (heyitsanthony) [inline Documentation/op-guide/v2-migration.md]: Transactions: In v3, etcd provides multi-key conditional transactions. Applications should use transactions in place of Compare-And-Swap operations.

File: Documentation/op-guide/v2-migration.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+- Flat key space: There are no directories in API v3, only keys. For example, "/a/b/c/" is a key. Range queries support getting all keys matching a given prefix.
+
+- Compacted responses. Operations like `Delete` no longer return previous values. To get the deleted value, a transaction can be used to atomically delete the key and get its value.
+
+- Lease instead of the TTL key. TTL is not bound to a key but a separate resource called lease. Multiple keys can be attached to a lease.
```
Comment (heyitsanthony) [inline Documentation/op-guide/v2-migration.md]: Leases: A replacement for v2 TTLs; the TTL is bound to a lease and keys attach to the lease. When the TTL expires, the lease is revoked and all attached keys are removed.

**Comment by heyitsanthony** [comment]
lgtm aside from final nits

**Comment by xiang90** [comment]
All fixed. Merging.

**Pull request merged**